src/Serialization/Rings.jl

```diff
@@ -47,6 +47,7 @@ type_params(x::T) where T <: IdealUnionType = TypeParams(T, base_ring(x))
 # exclude from ring union
 type_params(::ZZRing) = TypeParams(ZZRing, nothing)
 type_params(::ZZRingElem) = TypeParams(ZZRingElem, nothing)
+type_params(R::UniversalPolyRing) = TypeParams(UniversalPolyRing, coefficient_ring(R))
```

Comment:
I think we should also use `coefficient_ring` for the other "real" polynomial ring types, and then do
```suggestion
type_params(R::T) where T <: PolyRingUnionType = TypeParams(T, coefficient_ring(R))
```

Reply:
Yes, that's better.